client/core/core.go

```diff
@@ -1210,12 +1210,18 @@ func (c *Core) handleUnbookOrderMsg(dc *dexConnection, msg *msgjson.Message) err
 // listen monitors the DEX websocket connection for server requests and
 // notifications.
 func (c *Core) listen(dc *dexConnection) {
-	msgs := dc.MessageSource()
 	defer c.wg.Done()
+	msgs := dc.MessageSource()
 out:
 	for {
 		select {
-		case msg := <-msgs:
+		case msg, ok := <-msgs:
+			if !ok {
+				log.Warnf("Connection closed for %s.", dc.acct.url)
+				// TODO: remove from c.conns, and then?
```

Comment:
Maybe log an error or even panic if `c.ctx.Err() == nil`?

Reply:
I made some additional comments on this.